drivers/spi/spi_nrfx_spim.c

```diff
@@ -532,8 +532,7 @@ static int spim_nrfx_pm_action(const struct device *dev,
 	{								       \
 		struct spi_nrfx_data *dev_data = dev->data;		       \
 		int err;						       \
-		IRQ_CONNECT(NRFX_IRQ_NUMBER_GET(NRF_SPIM##idx),		       \
-			    DT_IRQ(SPIM(idx), priority),		       \
+		IRQ_CONNECT(DT_IRQN(SPIM(idx)), DT_IRQ(SPIM(idx), priority),   \
```

Comment:
ACK on this -- if you want to open a separate PR with this fix, we can merge it faster